### PR TITLE
Changes in SQ to make views run and extends supported measures

### PR DESCRIPTION
**Pull request by sns-seb**
This PR contains:
* mode of DAOs and Mapper from views to SQ
* addition of support for Views Component tree to a bunch of Measure Computation Steps
* fix of the loading of Views in SQ on Dory

### CODE REVIEW

File: server/sonar-server/src/main/java/org/sonar/server/computation/step/NewCoverageMeasuresStep.java [hunk 1]
```diff
@@ -141,13 +175,24 @@ public NewLineCoverageFormula() {
...
       super(batchReportReader,
         new NewCoverageInputMetricKeys(
           CoreMetrics.IT_COVERAGE_LINE_HITS_DATA_KEY, CoreMetrics.IT_CONDITIONS_BY_LINE_KEY, CoreMetrics.IT_COVERED_CONDITIONS_BY_LINE_KEY),
-        new NewCoverageOutputMetricKeys(
-          CoreMetrics.NEW_IT_LINES_TO_COVER_KEY, CoreMetrics.NEW_IT_UNCOVERED_LINES_KEY,
-          CoreMetrics.NEW_IT_CONDITIONS_TO_COVER_KEY, CoreMetrics.NEW_IT_UNCOVERED_CONDITIONS_KEY));
+        OUTPUT_METRIC_KEYS);
+    }
+
+    public static Iterable<Formula<?>> from(@Nullable BatchReportReader batchReportReader) {
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/computation/step/NewCoverageMeasuresStep.java]: MAJOR Remove usage of generic wildcard type. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1452)

File: server/sonar-server/src/main/java/org/sonar/server/computation/step/NewCoverageMeasuresStep.java [hunk 2]
```diff
@@ -103,13 +127,24 @@ public String getDescription() {
...
       super(batchReportReader,
         new NewCoverageInputMetricKeys(
           CoreMetrics.COVERAGE_LINE_HITS_DATA_KEY, CoreMetrics.CONDITIONS_BY_LINE_KEY, CoreMetrics.COVERED_CONDITIONS_BY_LINE_KEY),
-        new NewCoverageOutputMetricKeys(
-          CoreMetrics.NEW_LINES_TO_COVER_KEY, CoreMetrics.NEW_UNCOVERED_LINES_KEY,
-          CoreMetrics.NEW_CONDITIONS_TO_COVER_KEY, CoreMetrics.NEW_UNCOVERED_CONDITIONS_KEY));
+        OUTPUT_METRIC_KEYS);
+    }
+
+    public static Iterable<Formula<?>> from(@Nullable BatchReportReader batchReportReader) {
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/computation/step/NewCoverageMeasuresStep.java]: MAJOR Remove usage of generic wildcard type. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1452)

File: server/sonar-server/src/main/java/org/sonar/server/computation/step/NewCoverageMeasuresStep.java [hunk 3]
```diff
@@ -179,13 +224,24 @@ public NewItLinesCoverageFormula() {
...
       super(batchReportReader,
         new NewCoverageInputMetricKeys(
           CoreMetrics.OVERALL_COVERAGE_LINE_HITS_DATA_KEY, CoreMetrics.OVERALL_CONDITIONS_BY_LINE_KEY, CoreMetrics.OVERALL_COVERED_CONDITIONS_BY_LINE_KEY),
-        new NewCoverageOutputMetricKeys(
-          CoreMetrics.NEW_OVERALL_LINES_TO_COVER_KEY, CoreMetrics.NEW_OVERALL_UNCOVERED_LINES_KEY,
-          CoreMetrics.NEW_OVERALL_CONDITIONS_TO_COVER_KEY, CoreMetrics.NEW_OVERALL_UNCOVERED_CONDITIONS_KEY));
+        OUTPUT_METRIC_KEYS);
+    }
+
+    public static Iterable<Formula<?>> from(@Nullable BatchReportReader batchReportReader) {
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/computation/step/NewCoverageMeasuresStep.java]: MAJOR Remove usage of generic wildcard type. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1452)

**Comment by SonarTech** [comment]
SonarQube analysis reported 3 issues:
* MAJOR 3 major

Watch the comments in this conversation to review them.

**Comment by julienlancelot** [comment]
Some feedback, and LGTM !

**Comment by SonarTech** [comment]
SonarQube analysis reported 3 issues:
* MAJOR 3 major

Watch the comments in this conversation to review them.

**Comment by sns-seb** [comment]
regarding the DAO/Mapper discussion, didn't find any strong reason to take the required time to drop what exists, so I'll keep it
other comments applied

**Comment by julienlancelot** [comment]
Ok, no pb seb.

**Comment by SonarTech** [comment]
SonarQube analysis reported 3 issues:
* MAJOR 3 major

Watch the comments in this conversation to review them.

**Comment by SonarTech** [comment]
SonarQube analysis reported 3 issues:
* MAJOR 3 major

Watch the comments in this conversation to review them.

**Pull request merged**